java-frontend/src/test/java/org/sonar/java/ast/visitors/SyntaxHighlighterVisitorTest.java

```diff
@@ -192,13 +191,57 @@ void text_block() throws Exception {
     assertThatHasBeenHighlighted(componentKey, 8, 12, 10, 8, TypeOfText.STRING);
   }
 
-  private void scan(InputFile inputFile) {
-    JavaFrontend frontend = new JavaFrontend(new JavaVersionImpl(10), null, null, null, null, new JavaCheck[] {syntaxHighlighterVisitor});
-    frontend.scan(Collections.singletonList(inputFile), Collections.emptyList(), Collections.emptyList());
+  /**
+   * Java 15
+   */
+  @Test
+  void switch_expression() throws Exception {
+    this.eol = "\n";
+    InputFile inputFile = generateTestFile("src/test/files/highlighter/SwitchExpression.java");
+    scan(inputFile);
+
+    String componentKey = inputFile.key();
+    assertThatHasBeenHighlighted(componentKey, 9, 12, 9, 14, TypeOfText.KEYWORD); // yield true;
+    assertThatHasBeenHighlighted(componentKey, 13, 12, 13, 14, TypeOfText.KEYWORD); // yield false;
```

Comment:
It seems to work as expected, but we should probably use 9 for `startColumn` instead of 12 for consistency.